### PR TITLE
Refactor: FlowView class component to functional component

### PR DESCRIPTION
**Pull request by BastinJafari**
Refactor Flowview class component to functional component.

I got rid of:

```
    if (syncEvents.includes(event)) return this.syncFlow(flow);
    this.setState({ chart: flow });
```

because we need a new way of triggering a rerender (this.setState does it in the class, dispatching an action does it in the function). And we need a rerender everytime something changes with the chart/flow, so checking for specific types of events didn't seem to make sense to me.
I also think that the check didn't make sense in the class, because a rerender was triggered anyway with calling this.setState

### CODE REVIEW

File: src/views/FlowView.tsx [hunk 1]
```diff
@@ -148,174 +122,128 @@ class FlowView extends React.Component<any, any> {
...
     alert("Chart copied to clipboard as YAML");
   };
-
-  validateLink = ({ fromNodeId, toNodeId, fromPortId, toPortId }: any) => {
+  const validateLink = ({
+    fromNodeId,
+    toNodeId,
+    fromPortId,
+    toPortId,
+  }: any) => {
```
Comment (Roshanjossey) [inline src/views/FlowView.tsx:157]: Do we have to go with `any` here?

File: src/views/FlowView.tsx [hunk 2]
```diff
@@ -1,99 +1,90 @@
...
+    ...chart,
+    ...tooltipConfig,
+  };
+  const availableProperties = Store.getAvailableProperties();
+  const [connected, setConnected] = useState<boolean>(
+    Store.getConnectionStatus()
+  );
+  const [showOverlay, setShowOverlay] = useState<boolean>(false);
+
+  const actionCallbacks = Object.keys(actions).reduce((obj: any, key: any) => {
```
Comment (Roshanjossey) [inline src/views/FlowView.tsx:50]: I think you can get these types from the same package.

**Comment by Roshanjossey** [comment]
@BastinJafari, could you also update integration tests for FlowView as a part of this? When I tried last, one test case in it was failing.

Use a selector like `data-name` instead of classnames. 

Also, think about unit tests for functions in this FlowView.

File: src/views/FlowView.tsx (hunk 1 above)
Comment (BastinJafari) [reply src/views/FlowView.tsx:157]: I copied it from the original implementation. Changed the type to IOnLinkCompleteInput from the package

**Comment by BastinJafari** [comment]
I get an issue when trying to select the nodes via data-name:

image

Even though the attribute data-name is in the tree, Cypress doesn't seem to find it. Any idea how to fix that?

**Comment by Roshanjossey** [comment]
@BastinJafari, try the developer tools inside cypress test runner  (it's basically a browser). See if you see the element in the DOM

https://docs.cypress.io/guides/guides/debugging.html#Using-the-Developer-Tools

**Pull request merged**